src/components/PaginationFooter/Buttons/FirstPage.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { usePaginationContext } from '../context/PaginationProvider';
+import { ButtonTypes, VisibilityThresholds } from '../types';
+import { StyledButton } from '../styles';
+import usePaginationButton from '../use-pagination-button';
+import { DEFAULT_FONT_SIZE } from '../../../constants';
+
+const FirstPage = () => {
+  const { pageInfo, width } = usePaginationContext();
+  const onFirstPage = pageInfo.page === 0;
+  const { IconComponent, styledProps } = usePaginationButton({
+    type: ButtonTypes.FIRST_PAGE,
```

Comment:
fyi I think the members of an enum is suppose to be pascal case: https://www.typescriptlang.org/docs/handbook/enums.html